#! /bin/sh /usr/share/dpatch/dpatch-run
## call_ld_with_proper_flags.dpatch by Stefano Zacchiroli <zack@debian.org>
##
## All lines beginning with `## DP:' are a description of the patch.
## DP: Filter linkflags passed to ld by ocamlopt -pack and -output-obj
## DP: to remove the "-Wl," parts, which are only used when ocamlopt calls
## DP: gcc.

@DPATCH@
diff -urNad trunk~/Makefile trunk/Makefile
--- trunk~/Makefile	2007-03-05 10:18:22.000000000 +0100
+++ trunk/Makefile	2007-04-11 10:51:08.000000000 +0200
@@ -314,6 +314,7 @@
 partialclean::
 	rm -f ocaml toplevel/toplevellib.cma
 
+Wl = -Wl,
 # The configuration file
 
 utils/config.ml: utils/config.mlp config/Makefile
@@ -325,8 +326,8 @@
             -e 's|%%BYTELINK%%|$(BYTECC) $(BYTECCLINKOPTS)|' \
             -e 's|%%NATIVECC%%|$(NATIVECC) $(NATIVECCCOMPOPTS)|' \
             -e 's|%%NATIVELINK%%|$(NATIVECC) $(NATIVECCLINKOPTS)|' \
-            -e 's|%%PARTIALLD%%|$(PARTIALLD) $(NATIVECCLINKOPTS)|' \
-            -e 's|%%PACKLD%%|$(PARTIALLD) $(NATIVECCLINKOPTS) -o |' \
+            -e 's|%%PARTIALLD%%|$(PARTIALLD) $(subst $(Wl),,$(NATIVECCLINKOPTS))|' \
+            -e 's|%%PACKLD%%|$(PARTIALLD) $(subst $(Wl),,$(NATIVECCLINKOPTS)) -o |' \
             -e 's|%%BYTECCLIBS%%|$(BYTECCLIBS)|' \
             -e 's|%%NATIVECCLIBS%%|$(NATIVECCLIBS)|' \
             -e 's|%%RANLIBCMD%%|$(RANLIBCMD)|' \
